static/src/javascripts/bootstraps/enhanced/notifications.js

```diff
@@ -0,0 +1,180 @@
+                                sub = pushSubscription;
+                                modules.configureSubscribeTemplate();
+                            }
+                        );
+                    }
+                });
+            });
+        },
+
+        configureSubscribeTemplate: function () {
+            var subscribed = modules.checkSubscriptions(),
+                hasNoSubscriptions = modules.subscriptionsEmpty(),
+                src = template(subscribeTemplate, {
+                    className: hasNoSubscriptions ? '' : 'notifications-subscribe-link--has-subscriptions',
+                    text: subscribed ? 'Following story' : 'Follow story',
+                    imgMobile: svgs('notificationsExplainerMobile', ['mobile-only', 'notification-explainer']),
+                    imgDesktop: svgs('notificationsExplainerDesktop', ['hide-on-mobile', 'notification-explainer'])
+                });
+
+            fastdom.write(function () {
+                $('.js-liveblog-body').prepend(src);
+                mediator.emit('page:notifications:ready');
+            });
+
+            bean.on(document.body, 'click', '.js-notifications-subscribe-link', function () {
+                modules.buttonHandler();
+            });
+        },
+
+        setSubscriptionStatus: function (subscribed) {
+            var subscribeButton = $('.js-notifications-subscribe-link'),
+                subscribeLink = $('.notifications-subscribe-link--follow');
+            isSubscribed = subscribed;
+            if (subscribed) {
+                subscribeLink.addClass('notifications-subscribe-link--has-subscriptions');
+            }
+            subscribeButton[0].textContent = subscribed ?  'Following story' : 'Follow story';
+        },
+
+        buttonHandler: function () {
+            if (isSubscribed) {
+                modules.stopFollowingThis();
+            } else {
+                modules.subscribe();
+            }
+        },
+
+        subscribe: function () {
+            if (modules.subscriptionsEmpty()) {
+                reg.pushManager.subscribe({userVisibleOnly: true}).then(function (pushSubscription) {
+                    sub = pushSubscription;
+                    modules.followThis();
+                });
+            } else {
+                modules.followThis();
+            }
+        },
+
+
+        unSubscribe: function () {
+            if (modules.subscriptionsEmpty()) {
+                sub.unsubscribe().then(function () {
+                }).catch(function (error) {
+                    robust.log('07cm-frontendNotificatons', error);
+                });
+            }
+        },
+
+        subscriptionsEmpty: function () {
+            var subscriptions = userPrefs.get('subscriptions') || [];
+            return subscriptions.length ? false : true;
+        },
+
+        checkSubscriptions: function () {
+            var subscriptions = userPrefs.get('subscriptions') || [];
```

Comment:
I see this code quite a lot: `userPrefs.get('subscriptions') || []`

Perhaps we need a simple function called `getSubscriptions` to abstract that?